components/create_post/create_post.jsx

```diff
@@ -551,21 +560,50 @@ class CreatePost extends React.PureComponent {
         const {
             currentChannel: updateChannel,
             userIsOutOfOffice,
+            groupsWithAllowReference,
+            channelMemberCountsByGroup,
+            currentChannelMembersCount,
+            useGroupMentions
         } = this.props;
 
-        const currentMembersCount = this.props.currentChannelMembersCount;
         const notificationsToChannel = this.props.enableConfirmNotificationsToChannel && this.props.useChannelMentions;
+        let memberNotifyCount = 0;
+        let channelTimezoneCount = 0;
+        let mentions = [];
+        const notContainsAtChannel = !containsAtChannel(this.state.message);
+        if (this.props.enableConfirmNotificationsToChannel && notContainsAtChannel && useGroupMentions) {
+            // Groups mentioned in users text
+            mentions = groupsMentionedInText(this.state.message, groupsWithAllowReference);
+            if (mentions.length > 0) {
+                mentions = mentions.
+                    map((group) => {
+                        const mappedValue = channelMemberCountsByGroup[group.id];
+                        if (mappedValue && mappedValue.channel_member_count > Constants.NOTIFY_ALL_MEMBERS && mappedValue.channel_member_count > memberNotifyCount) {
+                            memberNotifyCount = mappedValue.channel_member_count;
+                            channelTimezoneCount = mappedValue.channel_member_timezones_count;
+                        }
+                        return `@${group.name}`;
+                    });
+            }
+        }
+
```

Comment:
There seems to be a hefty amount of logic in both create_post.jsx and create_comment.jsx, which makes sense but my only callout would be to consider if there's any opportunity for reuse?

Reply:
Agreed @mkraft  - but this PR never actually had changes to that file that was from Hosseins previous PR that this was built off of. Maybe we can make a ticket to cleanup and reduce duplication between the two files

Reply:
Yeah for sure. I was thinking this was new code. NVM then.